test/specs/default-controls.test.js

```diff
@@ -0,0 +1,371 @@
+      expect(instance.getDotIndexes(6, 2, 1, 'right')).toEqual([0, 2, 4, 5]);
+    });
+  });
+});
+
+describe('<NextButton />', () => {
+  describe('#nextButtonDisable for cellAlign = `left`', () => {
+    let instance;
+
+    beforeEach(async () => {
+      const wrapper = mount(<NextButton buttonDisable={() => null} />);
+      instance = wrapper.instance();
+    });
+
+    // testing smaller number of pages
+    it('should return true for slide index >= 3', () => {
+      const alignLeftIndex2 = {
+        wrapAround: false,
+        slidesToShow: 3,
+        currentSlide: 2,
+        cellAlign: 'left',
+        slideCount: 6
+      };
+      expect(instance.nextButtonDisabled(alignLeftIndex2)).toEqual(false);
+
+      // past the slide 3 should return true and disable the button
+      const testLeftIndex3 = {
+        wrapAround: false,
+        slidesToShow: 3,
+        currentSlide: 3,
+        cellAlign: 'left',
+        slideCount: 6
+      };
+      expect(instance.nextButtonDisabled(testLeftIndex3)).toEqual(true);
+
+      const testLeftIndex4 = {
+        wrapAround: false,
+        slidesToShow: 3,
+        currentSlide: 4,
+        cellAlign: 'left',
+        slideCount: 6
+      };
+      expect(instance.nextButtonDisabled(testLeftIndex4)).toEqual(true);
+    });
+
+    // testing extreme scenarios
+    it('should return true for slide index >= 4', () => {
+      const alignLeftIndex3 = {
+        wrapAround: false,
+        slidesToShow: 7,
+        currentSlide: 3,
+        cellAlign: 'left',
+        slideCount: 11
+      };
+      expect(instance.nextButtonDisabled(alignLeftIndex3)).toEqual(false);
+
+      const alignLeftIndex4 = {
+        wrapAround: false,
+        slidesToShow: 7,
+        currentSlide: 4,
+        cellAlign: 'left',
+        slideCount: 11
+      };
+      expect(instance.nextButtonDisabled(alignLeftIndex4)).toEqual(true);
+    });
+
+    // testing edge case scenarios
+    it('should return valid true or false value in edge cases', () => {
+      // when slideCount < slidesToShow and slideCount < slidesToShow
+      const alignLeftEdgeCase1 = {
+        wrapAround: false,
+        slidesToShow: 6,
+        currentSlide: 0,
+        cellAlign: 'left',
+        slideCount: 5
+      };
+      expect(instance.nextButtonDisabled(alignLeftEdgeCase1)).toEqual(true);
+
+      // when slideCount ===  slidesToShow
+      const alignLeftEdgeCase2 = {
+        wrapAround: false,
+        slidesToShow: 5,
+        currentSlide: 0,
+        cellAlign: 'left',
+        slideCount: 5
+      };
+      expect(instance.nextButtonDisabled(alignLeftEdgeCase2)).toEqual(true);
+    });
+  });
+
+  describe('#nextButtonDisable for cellAlign = `center`', () => {
+    let instance;
+
+    beforeEach(async () => {
+      const wrapper = mount(<NextButton buttonDisable={() => null} />);
+      instance = wrapper.instance();
+    });
+
+    // testing smaller number of pages
+    it('should return true for slide index 5', () => {
```

Comment:
same comment for this `it` statement as the one above. I think it would be more clear why we expect this test case if you add a little more info the statement.